src/EnergyPlus/PlantLoopHeatPumpEIR.cc

```diff
@@ -552,6 +643,90 @@ void EIRPlantLoopHeatPump::calcSourceSideHeatTransferASHP(EnergyPlusData &state)
     Real64 const CpSrc = Psychrometrics::PsyCpAirFnW(state.dataEnvrn->OutHumRat);
     Real64 const sourceMCp = this->sourceSideMassFlowRate * CpSrc;
     this->sourceSideOutletTemp = this->calcSourceOutletTemp(this->sourceSideInletTemp, this->sourceSideHeatTransfer / sourceMCp);
+    if (this->heatRecoveryAvailable && !this->heatRecoveryIsActive) {
+        // reset the HR report variables
+        this->heatRecoveryRate = 0.0;
+        this->heatRecoveryOutletTemp = this->heatRecoveryInletTemp;
+    }
+}
+
+void EIRPlantLoopHeatPump::calcHeatRecoveryHeatTransferASHP(EnergyPlusData &state)
+{
+    // energy balance on heat pump
+    this->heatRecoveryRate = this->calcQheatRecovery(this->loadSideHeatTransfer, this->powerUsage);
+    Real64 heatRecoverRateTot = this->heatRecoveryRate;
+
+    // calculate heat recovery side outlet conditions
+    auto &thisHeatRecoveryPlantLoop = state.dataPlnt->PlantLoop(this->heatRecoveryPlantLoc.loopNum);
+    Real64 const CpHR = FluidProperties::GetSpecificHeatGlycol(state,
+                                                               thisHeatRecoveryPlantLoop.FluidName,
+                                                               this->heatRecoveryInletTemp,
+                                                               thisHeatRecoveryPlantLoop.FluidIndex,
+                                                               "EIRPlantLoopHeatPump::calcHeatRecoveryHeatTransferASHP()");
+    Real64 const hRecoveryMCp = this->heatRecoveryMassFlowRate * CpHR;
+    if (this->heatRecoveryMassFlowRate > 0.0) {
+        this->heatRecoveryOutletTemp = this->calcHROutletTemp(this->heatRecoveryInletTemp, this->heatRecoveryRate / hRecoveryMCp);
+    } else {
+        this->heatRecoveryOutletTemp = this->heatRecoveryInletTemp;
+        this->heatRecoveryRate = 0.0;
+    }
+    // limit the HR HW outlet temperature to the maximum allowed (HW Recovery)
+    if (this->heatRecoveryOutletTemp > this->maxHeatRecoveryTempLimit) {
+        if (this->heatRecoveryInletTemp < this->maxHeatRecoveryTempLimit) {
+            this->heatRecoveryOutletTemp = this->maxHeatRecoveryTempLimit;
+            this->heatRecoveryRate = hRecoveryMCp * (this->heatRecoveryOutletTemp - this->heatRecoveryInletTemp);
+        } else {
+            this->heatRecoveryRate = 0.0;
+            this->heatRecoveryOutletTemp = this->heatRecoveryInletTemp;
+        }
+    }
```

Comment:
if the heat recovery outlet temperature exceeds the user specified limit, then reset the outlet temperature to the limit and recalculate the heat recovery heat transfer rate.